src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -0,0 +1,113 @@
+import Boom from 'boom';
+import { get, omit, pick } from 'lodash';
+
+import {
+  createFindQuery,
+  createFilterPath,
+  handleEsError,
+} from './lib';
+
+export class SavedObjectsClient {
+  constructor(kibanaIndex, request, callWithRequest) {
+    this._kibanaIndex = kibanaIndex;
+    this._request = request;
+    this._callWithRequest = callWithRequest;
+  }
+
+  async create(type, options = {}) {
+    const body = omit(options, 'id');
+
+    const response = await this._withKibanaIndex('index', {
+      type,
+      body
+    });
+
+    return Object.assign({
+      type: response._type,
+      id: response._id,
+      version: response._version
+    }, body);
+  }
+
+  async delete(type, id) {
+    const response = await this._withKibanaIndex('delete', {
+      type,
+      id,
+      refresh: 'wait_for'
+    });
+
+    if (get(response, 'found') === false) {
+      throw Boom.notFound();
+    }
+
+    return get(response, 'result') === 'deleted';
+  }
+
+  async find(options = {}) {
+    const { search, searchFields, type, fields } = options;
```

Comment:
It seems like this method would benefit from more destructuring/object literal use rather than using lodash methods to imperatively read the `options` and build the `esOptions`. I imagine this working like so:

```js
// defines the total structure of the options argument, allowing folks to easy see
// what options are accepted
const {
  search,
  searchFields,
  type,
  fields,
  perPage = 20, // defaults are defined in line, again describing the options argument to readers
  page = 1,
} = options;

// again, defines the total structure of the esOptions so that it is easier to see how options
// are converted to esOptions, without having to evaluate mutliple lines of imperative executions
// in your head
const esOptions = {
  type,
  filterPath: fields ? createFilterPath(fields) : undefined,
  size: perPage,
  from: perPage * (page - 1),
  body: createFindQuery({ search, searchFields, type })
};
```